### PR TITLE
WIP gdpr consent

### PR DESCRIPTION
**Pull request by hgwood**
See #36.

### CODE REVIEW

File: server/types.graphql [hunk 1]
```diff
@@ -95,4 +95,16 @@ type User @model {
 
   createdAt: DateTime!
   updatedAt: DateTime!
+
+  consentForPersonalData: Consent @relation(name: "UserConsent")
+}
+
+type Consent @model {
+  id: ID! @isUnique
+  user: [User!]! @relation(name: "UserConsent")
```
Comment (Errorname) [inline server/types.graphql]: "[User!]!" indique une liste d'utilisateurs. Ici on est sur une relation 1-1, donc à remplacer par "User"

File: server/src/auth0/auth0Authentication.js [hunk 2]
```diff
@@ -57,34 +57,45 @@ const getGraphcoolUser = (auth0UserId, api) =>
...
-						$email: String,
-						$name: String,
-						$given_name: String,
-						$family_name: String,
-						$picture: String,
-						$locale: String) {
+        mutation createUser(
+          $auth0UserId: String!,
+          $email: String,
+          $name: String,
```
Comment (Errorname) [inline server/src/auth0/auth0Authentication.js]: We also need to change every occurence of $given_name and $family_name to use $name.

I don't think $family_name is used, but $given_name is. Maybe we can do `$name.split(' ')[0]` to get it ?

File: src/services/search.js [hunk 3]
```diff
@@ -64,6 +64,15 @@ class Search {
   }
 }
 
-const search = new Search()
+const stubSearch = {
```
Comment (Errorname) [inline src/services/search.js]: I will try to do a PR dedicated to improve first-time contributors experience. I'll use these lines in it

File: src/scenes/Auth/Consent.jsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    return (
+      <div style={{ textAlign: 'center', marginTop: '2rem' }}>
+        <h1>Consent to store your personal data</h1>
+        <p>
+          This form gives you the opportunity to select what data FAQ may store about you. Please read carefully.
+          These options enhance the experience of FAQ but they are <em>not</em> required.
+          They can be changed later and at any time from your profile page.
+        </p>
+        <form>
+          <Checkbox checked={name} label="FAQ may store my name in order to display it against questions that I have asked, answers that I have written, and flags that I have placed." />
```
Comment (Errorname) [inline src/scenes/Auth/Consent.jsx]: The onChange function prop is missing on the Checkbox component.

```
<Checkbox checked={name} onChange={checked=>this.setState({name: checked}) ... />
```

File: server/types.graphql (hunk 1 above)
Comment (hgwood) [reply server/types.graphql]: Right. Actually I'm going to change the relationship to 1 user - n consents in order to have one row per consent. It's more flexible that way if we need to add some.

File: server/src/auth0/auth0Authentication.js (hunk 2 above)
Comment (hgwood) [reply server/src/auth0/auth0Authentication.js]: No we should not do that. Names are human text and should not be processed in any way. Either we use the name or we use the given name.

File: src/scenes/Auth/Consent.jsx (hunk 4 above)
Comment (hgwood) [reply src/scenes/Auth/Consent.jsx]: Right.

File: server/types.graphql (hunk 1 above)
Comment (Errorname) [reply server/types.graphql]: What's the benefit compared to adding an attribute to the Consent model ?

File: server/src/auth0/auth0Authentication.js (hunk 2 above)
Comment (Errorname) [reply server/src/auth0/auth0Authentication.js]: I agree

File: server/types.graphql (hunk 1 above)
Comment (hgwood) [reply server/types.graphql]: The schema does not change.

**Comment by hgwood** [comment]
We've agreed to shift priority from asking for consent to a complete and actionable profile page. So I'll close this for now, though I won't remove the branch.

**Pull request closed without merging**